pkg/suggestion/v1beta1/optuna/base_service.py

```diff
@@ -108,17 +109,50 @@ def _get_assignments_key(assignments):
 
     def _get_optuna_search_space(self):
         search_space = {}
+
         for param in self.search_space.params:
             if param.type == INTEGER:
-                search_space[param.name] = optuna.distributions.IntDistribution(
-                    int(param.min), int(param.max)
-                )
+                if param.distribution in [api_pb2.UNIFORM, None]:
+                    # Uniform integer distribution: samples integers between min and max.
+                    # If step is defined, use a quantized version.
+                    search_space[param.name] = optuna.distributions.IntDistribution(
+                        low=int(param.min),
+                        high=int(param.max),
+                        log=False,
+                        step=int(param.step) if param.step else None,
+                    )
+                elif param.distribution == api_pb2.LOG_UNIFORM:
+                    # Log-uniform integer distribution: used for exponentially varying integers.
+                    search_space[param.name] = optuna.distributions.IntDistribution(
+                        low=max(1, int(param.min)),
+                        high=int(param.max),
+                        log=True,
+                        step=1,
+                    )
+
             elif param.type == DOUBLE:
-                search_space[param.name] = optuna.distributions.FloatDistribution(
-                    float(param.min), float(param.max)
-                )
-            elif param.type == CATEGORICAL or param.type == DISCRETE:
+                if param.distribution in [api_pb2.UNIFORM, None]:
+                    # Uniform float distribution: samples values between min and max.
+                    # If step is provided, use a quantized version.
+                    search_space[param.name] = optuna.distributions.FloatDistribution(
+                        low=float(param.min),
+                        high=float(param.max),
+                        log=False,
+                        step=float(param.step) if param.step else None,
+                    )
+                elif param.distribution == api_pb2.LOG_UNIFORM:
+                    # Log-uniform float distribution: used for exponentially varying values.
+                    search_space[param.name] = optuna.distributions.FloatDistribution(
+                        low=max(1e-10, float(param.min)),
+                        high=float(param.max),
+                        log=True,
+                        step=None,
+                    )
+
+            elif param.type in {CATEGORICAL, DISCRETE}:
```

Comment:
Should we use array here ?
```suggestion
            elif param.type in [CATEGORICAL, DISCRETE]:
```